svd-rs/src/access.rs

```diff
@@ -0,0 +1,69 @@
+/// Defines access rights for fields on the device, though it may be specified at a
+/// higher level than individual fields.
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum Access {
+    /// Read access is permitted. Write operations have an undefined effect.
+    #[cfg_attr(feature = "serde", serde(rename = "read-only"))]
+    ReadOnly,
+
+    /// Read and write accesses are permitted.
+    #[cfg_attr(feature = "serde", serde(rename = "read-write"))]
+    ReadWrite,
+
+    /// Read access is always permitted.
+    /// Only the first write after a reset will affect the content.
+    /// Following writes have an undefined effect.
+    #[cfg_attr(feature = "serde", serde(rename = "read-writeOnce"))]
+    ReadWriteOnce,
+
+    /// Read operations have undefined results.
+    /// Only the first write after a reset will affect the content.
+    #[cfg_attr(feature = "serde", serde(rename = "writeOnce"))]
+    WriteOnce,
+
+    /// Read operations have an undefined result. Write access is permitted.
+    #[cfg_attr(feature = "serde", serde(rename = "write-only"))]
+    WriteOnly,
+}
+
+impl Access {
+    /// Whether the register/field is readable at least once.
+    pub fn can_read(self) -> bool {
+        matches!(self, Self::ReadOnly | Self::ReadWrite | Self::ReadWriteOnce)
+    }
+
+    /// Whether the register/field is writable at least once.
+    pub fn can_write(self) -> bool {
+        matches!(self, Self::ReadOnly)
+    }
+}
+
+impl Default for Access {
+    fn default() -> Self {
+        Self::ReadWrite
+    }
+}
+
+impl Access {
+    pub fn from_str(s: &str) -> Option<Self> {
```

Comment:
If serde wasn't optional, all of these from_str and to_str could be simplified.

I think the name should be changed though, I think parse_str suits better (and doesn't make clippy mad)

Reply:
Renamed `from_str` => `parse_str`, `to_str` => `as_str`.